src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs

```diff
@@ -1627,6 +1629,46 @@ public void MainWindowTitle_GetWithGui_ShouldRefresh_Windows()
             }
         }
 
+        [PlatformSpecific(TestPlatforms.Windows)] // it tests Windows implementation
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void RespondingIsRefreshedAfterEveryCallToRefresh()
+        {
+            // testing Process.Responding using a real unresponsive process would be very hard to do properly
+            // instead of this, we just test the implementation to ensure that #36768 is not coming back
+
+            using (Process process = CreateProcess())
+            {
+                process.Start();
+
+                try
+                {
+                    Assert.False(GetHaveResponding(process));
+
+                    Assert.True(process.Responding); // sets haveResponding to true
+                    Assert.True(GetHaveResponding(process));
+
+                    process.Refresh(); // sets haveResponding to false
```

Comment:
Given we're going down that route, wouldn't it make more sense to just make sure Refresh() invalidates all the right caches?

Reply:
Do you mean testing other fields as well? It's a good idea, I'll do that

Reply:
Yep, and perhaps renaming the test to something like `Refresh_Invalidates_All_Caches` (or something like that).

Reply:
@eiriktsarpalis  I've decided to give up on this idea for two reasons:

- the other two properties have existing tests: https://github.com/dotnet/runtime/blob/5d2490edcdd5165795f509ca5dfb98e6516e1e3a/src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs#L1564 and https://github.com/dotnet/runtime/blob/5d2490edcdd5165795f509ca5dfb98e6516e1e3a/src/libraries/System.Diagnostics.Process/tests/ProcessTests.cs#L1598
- the other two properties start returning values after the main windows gets loaded, to test them using the private fields I would have to basically copy-paste the code from the other existing tests that waits until the window gets loaded